File: frontend/src/api/permission.ts
import request from '@/utils/request'

export interface Permission {
  id?: number
  name: string
  code: string
  type?: number
  path?: string
  parentId?: number
  sortOrder?: number
  createdAt?: string
}

export const getPermissionList = () => {
  return request.get<Permission[]>('/v1/permissions')
}

export const getPermission = (id: number) => {
  return request.get<Permission>(`/v1/permissions/${id}`)
}

export const createPermission = (data: Permission) => {
  return request.post('/v1/permissions', data)
}

export const updatePermission = (id: number, data: Permission) => {
  return request.put(`/v1/permissions/${id}`, data)
}

export const deletePermission = (id: number) => {
  return request.delete(`/v1/permissions/${id}`)
}

File: backend/src/main/java/com/platform/entity/Permission.java
package com.platform.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.List;

@Data
@TableName("sys_permission")
public class Permission {

    @TableId(type = IdType.AUTO)
    private Long id;

    private String name;

    private String code;

    private Integer type;

    private String path;

    private Long parentId;

    private Integer sortOrder;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(exist = false)
    private List<Permission> children;
}
